### ISSUE TITLE
BorringSSL Handshake Failure 

### ISSUE THREAD

**Issue by WiseNN** [apple/swift-nio-ssl#420]
I am consistently getting SSL Handshake Failure, and am having issues troubleshooting this. Any help would be much appreciated. 

```swift
 handshakeFailed(NIOSSL.BoringSSLError.sslError([Error: 268436496 error:10000410:SSL routines:OPENSSL_internal:SSLV3_ALERT_HANDSHAKE_FAILURE at /Users/norriswise/Library/Developer/Xcode/DerivedData/ios-dc-bocetydygnmhxsdxqxaivnvasghk/SourcePackages/checkouts/swift-nio-ssl/Sources/CNIOBoringSSL/ssl/tls_record.cc:592])) file:[<unknown>] line:[0]]
```

**tls_record.cc**
```cc
 if (alert_level == SSL3_AL_FATAL) {
    OPENSSL_PUT_ERROR(SSL, SSL_AD_REASON_OFFSET + alert_descr);
    ERR_add_error_dataf("SSL alert number %d", alert_descr);
    *out_alert = 0;  // No alert to send back to the peer.
    return ssl_open_record_error;
  }
```

Swift Code: 
```swift
 var tlsConfig = TLSConfiguration.makeClientConfiguration()
        tlsConfig.maximumTLSVersion = .tlsv12
        tlsConfig.minimumTLSVersion = .tlsv11
        tlsConfig.cipherSuiteValues = NIOTLSCipher.allCases
        tlsConfig.certificateVerification = .noHostnameVerification
        tlsConfig.certificateChain = [.certificate(nioCert!)]
        let privateKeyNIO = try? NIOSSLPrivateKey.init(bytes: privateKeyByteAry, format: .der)
        tlsConfig.privateKey = NIOSSLPrivateKeySource.privateKey(privateKeyNIO!)
        
        let clientConfig = GRPCTLSConfiguration.makeClientConfigurationBackedByNIOSSL(configuration: tlsConfig, hostnameOverride: config.sniName)
        
        var gRPCLogger = Logger.init(label: "[[GRPC-LOGG]]", factory: StreamLogHandler.standardOutput(label:))
        gRPCLogger.logLevel = .debug
        
        if config.enableTLS {
            clientConnection = ClientConnection.usingTLS(with: clientConfig, on: eventLoopGroup)
                .withTLSCustomVerificationCallback({ certAry, verificationEventLoop in

                                      verificationEventLoop.succeed(.certificateVerified) //forcing success for testing purposes

                                  }                    
                             })
                .withBackgroundActivityLogger(gRPCLogger)
                .withErrorDelegate(self)
            clientConnection.withConnectionBackoff(retries: .upTo(3))
```

**Comment by WiseNN** [comment]
@Lukasa  I have been told from the backend guys that the certificates are not being seen; which I do think is correct from reviewing Charles. I see 3 server side certificates and 0 client side certificates. 




Wireshark Logs: 

Screenshot 1 - Wireshark
<img width="1344" alt="Screenshot 1" src="https://user-images.githubusercontent.com/17106171/217354686-951d2020-27f2-4aa9-ab6d-829a9811fb21.png">
<br></br>

Screenshot 2 - Wireshark
<img width="1349" alt="Screenshot 2" src="https://user-images.githubusercontent.com/17106171/217354807-31175023-dc21-46ad-b4e5-7e367893b103.png">
<br></br>

Screenshot 3 From Server Guy Wireshark
_basically saying we are missing a client certificate_ 
![Screenshot from Server Guy Wireshark (6)](https://user-images.githubusercontent.com/17106171/217354869-4462a342-702d-4a48-84c4-351ba23ad3f3.png)

**Comment by Lukasa** [comment]
Hmm, can you show me the Certificate messages coming from the client to the server? In the handshake you can see that the server requests a client cert using a `Certificate Request` message (same packet as the `Server Key Exchange` message in both cases), and then the client sends back a `Certificate` and `Certificate Verify` message in its next flight. That's where the client cert should be. Can you confirm the client cert is right, and that the server doesn't have any logs indicating that it's failing to verify it? It's notable that in your first screenshot the `Certificate` message is _extremely_ short (78 bytes), which seems likely to be wrong as it's barely enough space for the key.
